llvm/lib/Transforms/Scalar/NewGVN.cpp

```diff
@@ -1033,8 +1036,23 @@ PHIExpression *NewGVN::createPHIExpression(ArrayRef<ValPair> PHIOperands,
       return false;
     OriginalOpsConstant = OriginalOpsConstant && isa<Constant>(P.first);
     HasBackedge = HasBackedge || isBackedge(BB, PHIBlock);
-    return lookupOperandLeader(P.first) != I;
+
+    // Ignore phi arguments that are known to be equivalent to the phi. Only
+    // done for real PHIs (PHIOfOps may not converge).
```

Comment:
so what about phi-of-ops?
maybe open a new bug report with a trigger for phi-of-ops if possible